test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs

```diff
@@ -308,39 +328,42 @@ public void CanUpgradeRequestWithConnectionKeepAliveUpgradeHeader()
                     socket.Send(Encoding.ASCII.GetBytes("GET / HTTP/1.1\r\n\r\n"));
 
                     // Wait until connection is established
-                    Assert.True(await connectionStarted.WaitAsync(_semaphoreWaitTimeout));
+                    await connectionStartedTcs.Task.TimeoutAfter(TimeSpan.FromSeconds(10));
 
                     // Force a reset
                     socket.LingerState = new LingerOption(true, 0);
                 }
-
-                // Wait until connection error is logged
-                Assert.True(await connectionResetLogged.WaitAsync(_semaphoreWaitTimeout));
-
-                // Check the no log level higher than Debug was used for a reset between a requests
-                Assert.Equal(LogLevel.Debug, testSink.MaxLogLevel);
-
-                // Check for expected message
-                Assert.NotNull(testSink.ConnectionResetMessage);
-                Assert.Contains("reset", testSink.ConnectionResetMessage);
             }
+
+            mockLogger.Verify(logger => logger.Log(LogLevel.Debug, _connectionResetEventId, It.IsAny<object>(), null, It.IsAny<Func<object, Exception, string>>()));
         }
 
         [Fact]
-        public async Task ConnectionResetMidRequestIsLoggedAsInformation()
+        public async Task ConnectionResetMidRequestIsLoggedAsDebug()
```

Comment:
I think this test's name wasn't very accurate, since it's not really the reset that is logged as information. The information log is actually the bad request log message.